indico/modules/categories/controllers/management.py

```diff
@@ -224,3 +242,72 @@ def _process(self):
         subcategories = {category.id: category for category in self.category.children}
         for position, id_ in enumerate(request.json['categories'], 1):
             subcategories[id_].position = position
+
+
+class RHDeleteEvents(RHManageCategoryBase):
+    """Delete multiple events"""
+
+    def _checkParams(self):
+        RHManageCategoryBase._checkParams(self)
+        if request.form.get('all_selected') == 'true':
+            self.events = self.category.events
+        else:
+            event_ids = map(int, request.form.getlist('event_id'))
+            self.events = Event.query.with_parent(self.category).filter(Event.id.in_(event_ids)).all()
+
+    def _process(self):
+        is_submitted = 'submitted' in request.form
+        if not is_submitted:
+            return jsonify_template('events/management/delete_events.html', events=self.events)
+        for ev in self.events[:]:
+            delete_event(ev)
+        flash(ngettext('You have deleted one event', 'You have deleted {} events', len(self.events))
+              .format(len(self.events)), 'success')
+        return jsonify_data(flash=False, redirect=url_for('.manage_content', self.category))
+
+
+class RHSplitCategory(RHManageCategoryBase):
+    def _checkParams(self):
+        RHManageCategoryBase._checkParams(self)
+        self.category_events = {x.id: x for x in self.category.events}
+        if request.form.get('all_selected') == 'true':
+            self.event_ids = self.category_events.viewkeys()
+        else:
+            self.event_ids = set(map(int, request.form.getlist('event_id')))
+
+    def _process(self):
+        all_selected = not bool(self.category_events.viewkeys() - self.event_ids)
+        form = SplitCategoryForm(move_all=all_selected, event_id=self.event_ids)
+        if form.validate_on_submit():
+            selected_events = self.category_events.viewkeys() & self.event_ids
+            not_selected_events = self.category_events.viewkeys() - self.event_ids
+            if selected_events:
+                self._move_events(selected_events, form.first_category.data)
+            if not_selected_events:
+                self._move_events(not_selected_events, form.second_category.data)
+            return jsonify_data(flash=False, redirect=url_for('.manage_content', self.category))
```

Comment:
I think it is paramount to have a decent `flash(...)` message here, saying something like "Your events have been split between [category1](http://example.com/categ1) [category2](http://example.com/categ2)" or "Your events have been moved successfully" (in case it's just a single category).